cmd/generate/generate.go

```diff
@@ -139,15 +152,16 @@ func Validate(o *GenerateOptions) error {
 			o.format, strings.Join(validFormats, ", "))
 	}
 
-	if o.customEntriesPath == "" {
-		return nil
+	if !slices.Contains(validPlatformType, strings.ToLower(o.platformType)) {
+		return fmt.Errorf("invalid platform type %s, please use one of the following: %s", o.platformType,
+			strings.Join(validPlatformType, ", "))
 	}
 
-	if o.customEntriesFormat == "" {
+	if o.customEntriesPath != "" && o.customEntriesFormat == "" {
 		return fmt.Errorf("you must specify the --customEntriesFormat when using --customEntriesPath")
 	}
 
-	if !slices.Contains(validCustomEntriesFormats, o.customEntriesFormat) {
+	if o.customEntriesFormat != "" && !slices.Contains(validCustomEntriesFormats, o.customEntriesFormat) {
```

Comment:
why this change?

Reply:
this if was related so i made them to unrelated to each other
if u realize they was one after one

Reply:
this is unrelated to the "platform type" change, this should go in a separate commit